build.gradle

```diff
@@ -77,6 +77,24 @@ tasks.withType(Test).configureEach {
     useJUnitPlatform()
 }
 
+def setupPluginUpload = tasks.register("setupPluginUpload") {
+    doLast {
+        def key = System.getenv("GRADLE_PUBLISH_KEY")
+        def secret = System.getenv("GRADLE_PUBLISH_SECRET")
+
+        if (key == null || secret == null) {
+            throw new RuntimeException("PublishKey or PublishSecret are not defined from environment variables")
+        }
+
+        System.setProperty("gradle.publish.key", key)
+        System.setProperty("gradle.publish.secret", secret)
```

Comment:
Side note: It is disappointing that Gradle does not support an easy way to set credentials from environment variables which might be important for CI.
https://github.com/gradle/gradle/issues/1246